src/test/scala/scala/collection/immutable/TestLazyListExtensions.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection.immutable
+
+import org.junit.Assert._
+import org.junit.Test
+
+import next._
+
+class TestLazyListExtensions {
+
+  // This method will *not* terminate for non-cyclic infinite-sized collections.
+  def assertConstantMemory[T](xs: LazyList[T]): Unit =
+    // `force` does cycle detection, so if this terminates, the collection is
+    // either finite or a cycle
+    xs.force
+
+  @Test
+  def testEmpty1(): Unit = {
```

Comment:
I think these are good tests, but if they are all for `cycle`, which is not indicated anywhere in the method names. if we ever add another extension method, the test names will be a bit confusing. perhaps either `emptyCycle1` or `cycleEmpty1`.

also, small nit: I don't think the methods need to have `test` in their names.

Reply:
I took both of these suggestions.